jerry-core/parser/js/js-lexer.c

```diff
@@ -1476,17 +1475,17 @@ lexer_construct_literal_object (parser_context_t *context_p, /**< context */
     }
 
     if (literal_p->length == 4
-        && source_p[0] == 'e'
-        && source_p[3] == 'l'
-        && source_p[1] == 'v'
-        && source_p[2] == 'a')
+        && source_p[0] == LIT_CHAR_LOWERCASE_E
+        && source_p[3] == LIT_CHAR_LOWERCASE_L
+        && source_p[1] == LIT_CHAR_LOWERCASE_V
+        && source_p[2] == LIT_CHAR_LOWERCASE_A)
```

Comment:
Yes, when the first and last character matches, there is a high chance there will be a match.